core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -247,8 +247,12 @@ object TopicCommand extends Logging {
         val createResult = adminClient.createTopics(Collections.singleton(newTopic))
         createResult.all().get()
         println(s"Created topic ${topic.name}.")
-      } else {
-        throw new IllegalArgumentException(s"Topic ${topic.name} already exists")
+      } catch {
+        case e : ExecutionException =>
+          if (e.getCause == null)
+            throw e
+          if (!e.getCause.isInstanceOf[TopicExistsException] || !topic.ifTopicDoesntExist())
```

Comment:
nit: I had some trouble reading this, since it's two not operators with an OR statement. Would this be easier to read as `if e.getCause.isInstanceOf[TopicExistsException] && topic.ifTopicDoesntExist()` ? Note that [scala doesn't require a null check on the exception before calling isInstanceOf](https://stackoverflow.com/questions/14203201/does-scala-null-count-as-an-instance-of-another-type)

I think it's more conventional in Scala to have full if/else branches rather than guard statements

Reply:
Part of the issue here is I wanted to throw e, if the cause is null.. So I needed to check for that anyway. 

I was debating the `||` vs `&&` there. With the `if` you mentioned, I need to return out of the method, instead of throwing.. I prefer to keep this the way it is.

Reply:
I interpreted [~stanislavkozlovski]'s suggestion as being something like this:
```
          if (!(e.getCause.isInstanceOf[TopicExistsException] && topic.ifTopicDoesntExist()))
            throw e.getCause
```
So you wouldn't need to return out of the method in this case.

I do think that using AND here as suggested would be more intuitive.  The reason is because it's basically a special case we're handling here: we got TopicExistsException AND we had `--if-not-exists`.  Using the OR construct just feels unintuitive since we're basically taking the disjunction of the special case (enumerating all the ways we could not be in the special case.)

Anyway, I don't feel that strongly about it.  If you really want to keep it as is, that's OK.

Reply:
yeah.. I can move the `!` outside so its easier to reason.. Personally, I do fine with both, but given we have two similar feedback now. Happy to change.